### ISSUE TITLE
Enhance path_finding movement

### ISSUE THREAD

**Issue by derivator** [christopho/solarus#657]
It would be great if we could

* set the maximum distance and other parameters like refresh rate from lua, since 200 pixels manhattan distance is very short
* possibly use [hierarchical pathfinding](http://aigamedev.com/open/review/near-optimal-hierarchical-pathfinding/) (maybe with user defined large scale nodes) for large distances
* have callbacks for when no path can be found. The current behaviour of just walking randomly might not fit every use case and doesn't seem to be documented. (The same goes for reaching the target, we might want to do something else than switching to random walking)
* have PathFinding exposed to lua so people can do their own things with it

Might not seem very useful at the moment, but it would help for people who like to take a simulationist approach to their games. (large maps with fully simulated enemies with complex goals, which gives me another idea: teletransporters that can be used by enemies, which the pathfinding could take into account)

**Comment by wrightmat** [comment]
Yes - can't agree with this more! This is really necessary for complex NPC routines as well.
